components/script/dom/performance.rs

```diff
@@ -139,7 +136,8 @@ pub struct Performance {
     buffer: DomRefCell<PerformanceEntryList>,
     observers: DomRefCell<Vec<PerformanceObserver>>,
     pending_notification_observers_task: Cell<bool>,
-    navigation_start_precise: u64,
+    #[no_trace]
+    time_origin: CrossProcessInstant,
```

Comment:
I think here we can add https://html.spec.whatwg.org/multipage/#concept-settings-object-time-origin

The question that remains to me is: how do we share this per [`environment-settings-object`](https://html.spec.whatwg.org/multipage/webappapis.html#environment-settings-object)?

Reply:
I've added the spec link. I think this is already close to the concept of the `timeOrigin` as it is set per-`Worker` and also per `GlobalScope` in the page. Any further improvement should probably come in followups.

Reply:
Right, thanks for clarifying.